From a684cbcafc67cd5160d89018d6d1b39f1318446b Mon Sep 17 00:00:00 2001
From: Lim Siew Hoon <siew.hoon.lim@intel.com>
Date: Tue, 2 Nov 2021 23:39:23 +0800
Subject: [PATCH] msdkvpp: Add 12bit formats

Add 12bit formats for different chroma samplings at sink pad and
src pad, including P012_LE, Y212_LE and Y412_LE.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/1218>

Signed-off-by: Mengkejiergeli Ba <mengkejiergeli.ba@intel.com>
Signed-off-by: Lim Siew Hoon <siew.hoon.lim@intel.com>
---
 sys/msdk/gstmsdkvpp.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/sys/msdk/gstmsdkvpp.c b/sys/msdk/gstmsdkvpp.c
index 0439e49dd..50687e2f3 100644
--- a/sys/msdk/gstmsdkvpp.c
+++ b/sys/msdk/gstmsdkvpp.c
@@ -71,11 +71,14 @@
 #endif
 
 #if (MFX_VERSION >= 2004)
-#define EXT_SINK_FORMATS        ", RGB16, Y410, Y210"
-#define EXT_SRC_FORMATS         ", YV12, Y410, Y210, RGBP, BGRP"
+#define EXT_SINK_FORMATS        ", RGB16, Y410, Y210, P012_LE, Y212_LE, Y412_LE"
+#define EXT_SRC_FORMATS         ", YV12, Y410, Y210, RGBP, BGRP, P012_LE, Y212_LE, Y412_LE"
 #elif (MFX_VERSION >= 1032)
-#define EXT_SINK_FORMATS        ", RGB16, Y410, Y210"
-#define EXT_SRC_FORMATS         ", YV12, Y410, Y210"
+#define EXT_SINK_FORMATS        ", RGB16, Y410, Y210, P012_LE, Y212_LE, Y412_LE"
+#define EXT_SRC_FORMATS         ", YV12, Y410, Y210, P012_LE, Y212_LE, Y412_LE"
+#elif (MFX_VERSION >= 1031)
+#define EXT_SINK_FORMATS        ", RGB16, Y410, Y210, P012_LE, Y212_LE, Y412_LE"
+#define EXT_SRC_FORMATS         ", Y410, Y210, P012_LE, Y212_LE, Y412_LE"
 #elif (MFX_VERSION >= 1028)
 #define EXT_SINK_FORMATS        ", RGB16, Y410, Y210"
 #define EXT_SRC_FORMATS         ", Y410, Y210"
-- 
2.33.1

